server: clarify the unix socket for secondary tenants

knz opened the pull request:
Since there is now a single SQL listener for shared-process tenant servers, there's no need for concern about the unix socket path.

Release note: None
Epic: CRDB-14537

knz committed:
server: clarify the unix socket for secondary tenants

Since there is now a single SQL listener for shared-process tenant
servers, there's no need for concern about the unix socket path.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/96169)
<!-- Reviewable:end -->

stevendanna reviewed (commented):
Overall this seems reasonable to me.  Do we have a way to add a test to this to ensure we can connect to the tenant over the unix socket?

knz committed:
democluster: avoid a bool for a parameter

This introduces an enum type to replace true/false for the tenant
selection.

Release note: None

knz reviewed (commented):
>  Do we have a way to add a test to this to ensure we can connect to the tenant over the unix socket?

Good call. I had forgotten. Thanks - Done.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96169#-:-NN2xN_02dxPq7EV_HcC:b-8frhgl)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss and @stevendanna)

___
*[`pkg/cli/democluster/demo_cluster.go` line 585 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96169#-NN0ybY9hTmnPtzW3M9p-r1-585:-NN2yw4099GzS8o50KKm:b-21bxqc) ([raw file](https://github.com/cockroachdb/cockroach/blob/82b5c1ab1c69da2e6160ea530b9871835b92f318/pkg/cli/democluster/demo_cluster.go#L585)):*
<details><summary><i>Previously, stevendanna (Steven Danna) wrote…</i></summary><blockquote>

[style,musing] We use bool arguments with comments in a lot of places, but sometimes I wish we used a custom type with some constants instead. Anyway, doesn't need to change here.
</blockquote></details>

It's OK to ask. Done, added a new commit.


<!-- Sent from Reviewable.io -->

knz commented:
TFYR!

bors r=stevendanna

knz commented:
bors r=stevendanna,rafiss

knz commented:
bors r=stevendanna,rafiss
